hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/ITTestDataStreamWrite.java

```diff
@@ -167,6 +161,8 @@ public void testWriteMergeOnReadWithCompaction(String indexType) throws Exceptio
     conf.setString(FlinkOptions.INDEX_TYPE, indexType);
     conf.setInteger(FlinkOptions.BUCKET_INDEX_NUM_BUCKETS, 4);
     conf.setInteger(FlinkOptions.COMPACTION_DELTA_COMMITS, 1);
+    // use synchronized compaction to ensure flink job finishing with compaction completed.
+    conf.set(FlinkOptions.COMPACTION_ASYNC_ENABLED, false);
```

Comment:
Flink does not have inline compaction.

Reply:
Writing pipeline in this test is customized, not same with that in `HoodieTableSink`. Compact operator is added if the table type is MOR, see Line 285.